### PR TITLE
[MB-4229] Hide back btn and add 'finish later' btn to review page

### PR DESCRIPTION
**Pull request by amitch23**
## Description

Removes "back" button and adds "Finish Later" button to the review page. Since there are a handful of other tickets to deal with routing and buttons on the whole flow (see tickets tagged with the epic "routing updates" - https://dp3.atlassian.net/browse/MB-3849), I adapted the Wizard Page to take props to hide the back btn and to show the "finish later" button. I thought this best in order to roll out the changes iteratively.

## Review notes

The desktop view shows the cancel button still. Since we will be removing the cancel button in a future ticket, it seems like that work will be taken care of outside of the scope of this ticket.

## Setup

See the review page and ensure that there is no back button, and the "finish later" routes to the home page.

## Code Review Verification Steps

* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/secure/RapidBoard.jspa?rapidView=1&projectKey=MB&modal=detail&selectedIssue=MB-4229&quickFilter=60) for this change

## Screenshots

Screen Shot 2020-10-01 at 11 01 23 AM
Screen Shot 2020-10-01 at 11 01 33 AM

### CODE REVIEW

File: src/shared/WizardPage/index.jsx [hunk 1]
```diff
@@ -123,6 +125,17 @@ export class WizardPage extends Component {
                 Cancel
               </button>
             )}
+            {showFinishLaterBtn && (
+              <button
+                type="button"
+                className="usa-button usa-button--unstyled"
+                onClick={this.cancelFlow}
+                disabled={false}
+                data-testid="wizardFinishLaterButton"
```
Comment (kimallen) [inline src/shared/WizardPage/index.jsx:134]: I see a lot of the `data-testid` values in kebab case in the cypress tests, and then here you are using camelCase.  Is camelCase the preferred standard?

File: cypress/integration/mymove/hhg.js [hunk 2]
```diff
@@ -233,12 +232,7 @@ function customerSetsUpAnHHGMove() {
...
 }
 
-function customerAddsAnotherShipment() {
-  cy.get('button[data-testid="wizardBackButton"]').should('be.enabled').click();
-  customerSetsUpAnHHGMove();
-}
-
-function customerReviewsMoveDetails() {
+function customerReviewsMoveDetailsAndEditsHHG() {
   cy.get('[data-testid="review-move-header"]').contains('Review your details');
```
Comment (kimallen) [inline cypress/integration/mymove/hhg.js:236]: I'm wondering if we want a test to click the `Finish Later` button and take them to the home page and then have them click `Review and submit` to get to the Review page again.  I think we are trying to put more emphasis on client tests and focus mostly on the happy path in integration tests, so I'm not sure.

**Review by kimallen — CHANGES_REQUESTED** [review]
Code looks good and works as expected!

It looks like from [this Abstract](https://app.abstract.com/projects/1f6c1fd0-88a2-11e9-aaf7-a7ce5896598b/branches/master/commits/latest/files/4794a2f1-15d3-42e5-aa8f-be21f9ede9ce/layers/7A3BED69-D33D-4A02-BDEF-81B6B21B3BFE?collectionId=72626442-df23-421f-bd5c-3ae6ade7d534&collectionLayerId=afa6c581-79e8-452b-8683-f09188d2bc9c) that we want the `Next` and `Finish later` buttons to be stacked when in phone view. 
image
 That and making 'later' lowercase are the only blocking changes.

File: src/shared/WizardPage/index.jsx [hunk 3]
```diff
@@ -123,6 +125,17 @@ export class WizardPage extends Component {
...
+              >
+                Finish Later
```
Comment (kimallen) [inline src/shared/WizardPage/index.jsx:136]: later should be lower case

File: src/shared/WizardPage/index.jsx (hunk 1 above)
Comment (amitch23) [reply src/shared/WizardPage/index.jsx:134]: I followed the pattern in this component for the buttons' data-testid values to be camelCase. I also see both, and I don't know which is the preferred standard. I asked in slack and will update with the answer here.

File: cypress/integration/mymove/hhg.js (hunk 2 above)
Comment (amitch23) [reply cypress/integration/mymove/hhg.js:236]: That's a good idea. :) I'll give that a go

File: src/shared/WizardPage/index.jsx (hunk 1 above)
Comment (amitch23) [reply src/shared/WizardPage/index.jsx:134]: @suzubara says "I don’t think we have discussed or documented. I would throw a vote out there for camelCase to be consistent with JS var naming :shrug:", How about we stick to camelCase?

**Comment by amitch23** [comment]
@Shkeating @suzubara I'm having trouble with the formatting and display of the buttons to match the spec and was hoping you could offer some insight and help. I'll list the issues here, referencing the screenshots in the PR description:
1. Because the wizard buttons are inside a `grid-row`, it seems like the spacing of the buttons on the right is off and doesn't align with the "add a shipment" button. Moving them out of a row and into a `grid-col-row` fixes it, but then the buttons are not horizontally aligned on desktop view, but rather, vertically aligned. Not a good look :/
2. The "finish later" button size does not match the "next" button, and it's much "smaller", as compared to the screenshot of the buttons Kim posted above.
3. I'm not sure how to handle the margin between the "next" and "finish later" buttons and the "add shipment button" section. I adapted the existing style applied to the div on the wizard page (changed from .5rem to 2rem) but am not sure if this is the best practice way to handle this. I also don't know what the units in abstract represent - is that 6 px? Or 6 units of a pixel measurement?

File: src/shared/WizardPage/index.jsx [hunk 4]
```diff
@@ -77,54 +79,65 @@ export class WizardPage extends Component {
...
-                onClick={this.cancelFlow}
-                disabled={false}
-                data-testid="wizardCancelButton"
-              >
-                Cancel
-              </button>
-            )}
-          </div>
+        <div className="grid-row" style={{ marginTop: '2rem' }}>
+          {/* <div className="grid-col-10 text-right margin-top-6 margin-left-neg-1 tablet:margin-top-3 display-flex"> */}
```
Comment (suzubara) [inline src/shared/WizardPage/index.jsx:83]: can we delete this line altogether?

File: src/shared/WizardPage/index.css [hunk 5]
```diff
@@ -24,3 +24,15 @@
 .hide-btn {
   display: none;
 }
+
```
Comment (suzubara) [inline src/shared/WizardPage/index.css:27]: I don't love this but acknowledge it's prob the easiest way to apply these styles until the wizard nav can be refactored 🧹

**Review by suzubara — APPROVED** [review]
looks good!

**Review by kimallen — APPROVED** [review]
Changes approved!

**Review by Shkeating — APPROVED** [review]
lookin good!

**Pull request merged**